apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java

```diff
@@ -637,29 +641,29 @@ void testOutputDevices() throws InterruptedException {
                 int[] channelCounts = deviceInfo.getChannelCounts();
                 numTested++;
                 // Always test mono and stereo.
-                testOutputDeviceCombo(id, deviceType, 1, 0, false);
-                testOutputDeviceCombo(id, deviceType, 2, 0, false);
-                testOutputDeviceCombo(id, deviceType, 2, 1, false);
+                testOutputDeviceCombo(id, deviceType, 1, 0, 0);
+                testOutputDeviceCombo(id, deviceType, 2, 0, 0);
+                testOutputDeviceCombo(id, deviceType, 2, 0, 1);
                 if (channelCounts.length > 0) {
                     for (int numChannels : channelCounts) {
                         // Test higher channel counts.
                         if (numChannels > 2) {
                             log("numChannels = " + numChannels + "\n");
                             for (int channel = 0; channel < numChannels; channel++) {
-                                testOutputDeviceCombo(id, deviceType, numChannels, channel, false);
+                                testOutputDeviceCombo(id, deviceType, numChannels, 0, channel);
                             }
                         }
                     }
                 }
                 int[] channelMasks = deviceInfo.getChannelMasks();
                 if (channelMasks.length > 0) {
                     for (int channelMask : channelMasks) {
-                        int nativeChannelMask = channelMask >> 2;
-                        // Test higher channel counts.
+                        int nativeChannelMask =
+                                convertJavaChannelMaskToNativeChannelMask(channelMask);
                         log("nativeChannelMask = " + nativeChannelMask + "\n");
                         int channelCount = Integer.bitCount(nativeChannelMask);
                         for (int channel = 0; channel < channelCount; channel++) {
-                            testOutputDeviceCombo(id, deviceType, nativeChannelMask, channel, true);
+                            testOutputDeviceCombo(id, deviceType, nativeChannelMask, 0, channel);
```

Comment:
The parameters seem out of order. Maybe:
    
    testOutputDeviceCombo(id, deviceType, 0, nativeChannelMask, channel);